File: src/data/strokes.js
/**
 * Stroke order guide data for Devanagari vowels.
 *
 * Each entry is an array of stroke guides. A guide marks where a stroke
 * begins and which direction it travels, so the guided-tracing mode can
 * render numbered arrows on the canvas.
 *
 * Coordinates are normalised to a [0, 1] square (multiply by canvas size
 * to get pixel positions). Angles follow standard math convention adapted
 * for screen coordinates: 0 = right, 90 = down, 180 = left, 270 = up.
 *
 * The character outline itself comes from font rendering on the canvas,
 * not from these points — so the guides only need to mark stroke starts
 * and directions, not trace the full path.
 *
 * @typedef {Object} StrokeGuide
 * @property {number} x - Normalised x of stroke start (0–1).
 * @property {number} y - Normalised y of stroke start (0–1).
 * @property {number} angle - Direction the stroke travels (degrees).
 */

/** @type {Record<string, StrokeGuide[]>} */
export const strokeGuides = {
  'अ': [
    { x: 0.48, y: 0.18, angle: 225 },
    { x: 0.62, y: 0.18, angle: 90 },
    { x: 0.18, y: 0.18, angle: 0 },
  ],
  'आ': [
    { x: 0.38, y: 0.18, angle: 225 },
    { x: 0.50, y: 0.18, angle: 90 },
    { x: 0.72, y: 0.18, angle: 90 },
    { x: 0.15, y: 0.18, angle: 0 },
  ],
  'इ': [
    { x: 0.45, y: 0.22, angle: 135 },
    { x: 0.40, y: 0.10, angle: 330 },
    { x: 0.20, y: 0.18, angle: 0 },
  ],
  'ई': [
    { x: 0.45, y: 0.22, angle: 135 },
    { x: 0.42, y: 0.62, angle: 90 },
    { x: 0.20, y: 0.18, angle: 0 },
  ],
  'उ': [
    { x: 0.35, y: 0.22, angle: 90 },
    { x: 0.30, y: 0.18, angle: 0 },
  ],
  'ऊ': [
    { x: 0.32, y: 0.22, angle: 90 },
    { x: 0.50, y: 0.60, angle: 90 },
    { x: 0.22, y: 0.18, angle: 0 },
  ],
  'ऋ': [
    { x: 0.30, y: 0.22, angle: 45 },
    { x: 0.35, y: 0.45, angle: 135 },
    { x: 0.62, y: 0.18, angle: 90 },
    { x: 0.18, y: 0.18, angle: 0 },
  ],
  'ए': [
    { x: 0.35, y: 0.28, angle: 200 },
    { x: 0.62, y: 0.18, angle: 90 },
    { x: 0.18, y: 0.18, angle: 0 },
  ],
  'ऐ': [
    { x: 0.35, y: 0.28, angle: 200 },
    { x: 0.38, y: 0.08, angle: 330 },
    { x: 0.62, y: 0.18, angle: 90 },
    { x: 0.18, y: 0.18, angle: 0 },
  ],
  'ओ': [
    { x: 0.32, y: 0.28, angle: 200 },
    { x: 0.68, y: 0.06, angle: 90 },
    { x: 0.58, y: 0.18, angle: 90 },
    { x: 0.15, y: 0.18, angle: 0 },
  ],
  'औ': [
    { x: 0.32, y: 0.28, angle: 200 },
    { x: 0.35, y: 0.08, angle: 330 },
    { x: 0.68, y: 0.06, angle: 90 },
    { x: 0.58, y: 0.18, angle: 90 },
    { x: 0.15, y: 0.18, angle: 0 },
  ],
}

File: app/lib/data/strokes.dart
// Stroke order guide data for Devanagari vowels.
// Each entry marks where a stroke starts and which direction it travels,
// so the guided-tracing mode can render numbered arrows on the canvas.
// Coordinates are normalised to [0, 1]. Angles: 0=right, 90=down, 180=left, 270=up.

import 'dart:math' as math;

/// A single stroke-order guide marker.
class StrokeGuide {
  final double x;
  final double y;
  final double angle;

  const StrokeGuide(this.x, this.y, this.angle);

  double get angleRad => angle * math.pi / 180;
}

/// Stroke order guides keyed by character.
const Map<String, List<StrokeGuide>> strokeGuides = {
  'अ': [
    StrokeGuide(0.48, 0.18, 225),
    StrokeGuide(0.62, 0.18, 90),
    StrokeGuide(0.18, 0.18, 0),
  ],
  'आ': [
    StrokeGuide(0.38, 0.18, 225),
    StrokeGuide(0.50, 0.18, 90),
    StrokeGuide(0.72, 0.18, 90),
    StrokeGuide(0.15, 0.18, 0),
  ],
  'इ': [
    StrokeGuide(0.45, 0.22, 135),
    StrokeGuide(0.40, 0.10, 330),
    StrokeGuide(0.20, 0.18, 0),
  ],
  'ई': [
    StrokeGuide(0.45, 0.22, 135),
    StrokeGuide(0.42, 0.62, 90),
    StrokeGuide(0.20, 0.18, 0),
  ],
  'उ': [
    StrokeGuide(0.35, 0.22, 90),
    StrokeGuide(0.30, 0.18, 0),
  ],
  'ऊ': [
    StrokeGuide(0.32, 0.22, 90),
    StrokeGuide(0.50, 0.60, 90),
    StrokeGuide(0.22, 0.18, 0),
  ],
  'ऋ': [
    StrokeGuide(0.30, 0.22, 45),
    StrokeGuide(0.35, 0.45, 135),
    StrokeGuide(0.62, 0.18, 90),
    StrokeGuide(0.18, 0.18, 0),
  ],
  'ए': [
    StrokeGuide(0.35, 0.28, 200),
    StrokeGuide(0.62, 0.18, 90),
    StrokeGuide(0.18, 0.18, 0),
  ],
  'ऐ': [
    StrokeGuide(0.35, 0.28, 200),
    StrokeGuide(0.38, 0.08, 330),
    StrokeGuide(0.62, 0.18, 90),
    StrokeGuide(0.18, 0.18, 0),
  ],
  'ओ': [
    StrokeGuide(0.32, 0.28, 200),
    StrokeGuide(0.68, 0.06, 90),
    StrokeGuide(0.58, 0.18, 90),
    StrokeGuide(0.15, 0.18, 0),
  ],
  'औ': [
    StrokeGuide(0.32, 0.28, 200),
    StrokeGuide(0.35, 0.08, 330),
    StrokeGuide(0.68, 0.06, 90),
    StrokeGuide(0.58, 0.18, 90),
    StrokeGuide(0.15, 0.18, 0),
  ],
};
